spec/unit/action/runner_spec.rb

```diff
@@ -170,6 +171,58 @@ def call
     end
   end
 
+  describe "configuring git credentials" do
+    it 'errors if both token and key paths are passed' do
+      runner = described_class.new({ 'oauth-token': '/nonexistent',
+                                     'private-key': '/also/fake' }, %w[args yes], action_class)
+      expect{ runner.call }.to raise_error(R10K::Error, /Cannot specify both/)
+    end
+
+    it 'saves the sshkey path in settings hash' do
+      runner = described_class.new({ 'private-key': '/my/ssh/key' }, %w[args yes], action_class)
+      runner.call
+      expect(runner.instance.settings[:git][:private_key]).to eq('/my/ssh/key')
+    end
+
+    it 'overrides per-repo sshkey in settings hash' do
+      runner = described_class.new({ config: "spec/fixtures/unit/action/r10k_creds.yaml",
+                                     'private-key': '/my/ssh/key' },
+                                     %w[args yes],
+                                     action_class)
+      runner.call
+      expect(runner.instance.settings[:git][:private_key]).to eq('/my/ssh/key')
+      expect(runner.instance.settings[:git][:repositories].count).to eq(2)
+      runner.instance.settings[:git][:repositories].each do |repo_settings|
+        expect(repo_settings[:private_key]).to eq('/my/ssh/key')
+      end
+    end
+
+    it 'saves the token path in settings hash' do
+      runner = described_class.new({ 'oauth-token': '/my/token/path' }, %w[args yes], action_class)
+      runner.call
+      expect(runner.instance.settings[:git][:oauth_token]).to eq('/my/token/path')
+    end
+
+    it 'saves the stdin placeholder to the settings hash' do
+      runner = described_class.new({ 'oauth-token': '-' }, %w[args yes], action_class)
+      runner.call
+      expect(runner.instance.settings[:git][:oauth_token]).to eq('-')
+    end
```

Comment:
Is this a meaningful test? It looks like it's identical to the regular token path test, just with a different value for the path.

Reply:
Probably not. I was sorta thinking to leave it for "tests as documentation", but I don't feel strongly.